### PR TITLE
Expose System.Media.SoundPlayer in System.Windows.Extensions

### PR DESCRIPTION
**Pull request by maryamariyan**
Reopening https://github.com/dotnet/corefx/pull/32505 on my own fork. It contains:
- Original commits @hughbe had on the PR
- Feedbacks marked as resolved in that PR

There's nothing blocking putting this into corefx at this moment.
Original reason we closed it earlier was due to this comment:
https://github.com/dotnet/corefx/pull/32505#issuecomment-437122365

After merging this PR, and exposing SoundPlayer in `System.Windows.Extensions`, we'll need to remove System.Media.SoundPlayer.dll and move its consumers (`System.Windows.Forms` and `PresentationFramework`) to use `System.Windows.Extensions.dll` instead.

I'll be in contact with @AdamYoblick so we can make sure we have all of that done after merging this PR.

@stephentoub should the remaining comments in the PR also be addressed in the first PR here or they can be done in a follow up PR?

cc: @danmosemsft @ericstj

### CODE REVIEW

File: src/System.Windows.Extensions/src/System.Windows.Extensions.csproj [hunk 1]
```diff
@@ -48,17 +72,28 @@
   <ItemGroup Condition="'$(TargetGroup)' == 'netcoreapp'">
     <Reference Include="System.Buffers" />
     <Reference Include="System.Collections.NonGeneric" />
+    <Reference Include="System.Collections.NonGeneric" />
```
Comment (danmosemsft) [inline src/System.Windows.Extensions/src/System.Windows.Extensions.csproj]: duplicate entry

File: src/System.Windows.Extensions/src/System.Windows.Extensions.csproj [hunk 2]
```diff
@@ -48,17 +72,27 @@
...
     <Reference Include="System.Runtime.InteropServices" />
+    <Reference Include="System.Runtime" />
     <Reference Include="System.Security.Cryptography.X509Certificates" />
     <Reference Include="System.Text.RegularExpressions" />
+    <Reference Include="System.Threading.Thread" />
     <Reference Include="System.Threading" />
   </ItemGroup>
-</Project>
\ No newline at end of file
+  <ItemGroup>
```
Comment (ericstj) [inline src/System.Windows.Extensions/src/System.Windows.Extensions.csproj]: These need a condition, better yet put them in the same itemgroup as the other compile items immediately following `System\Drawing\Printing\MarginsConverter.cs`

**Comment by maryamariyan** [comment]
I'm also adding more test coverage. 
There's no test coverage for the portion of code @stephentoub commented on having to do with aborting thread which would throw PNSE on .NET Core.

**Comment by maryamariyan** [comment]
@dotnet-bot test Linux arm64 Release Build

**Comment by maryamariyan** [comment]
The consumers of SoundPlayer (`System.Windows.Forms` and `PresentationFramework`) are using the .NET Core configuration and with the same implementation as in this PR (including thread Abort calls).
So this means merging the PR as is won't introduce new risk to existing consumers. Then the remaining work can be continued in a separate PR. 
That could also be safer since if there is any issues with the changes we can roll back easier.

**Comment by maryamariyan** [comment]
@dotnet/dnceng Linux arm64 release build failing on timeout

```
14:15:29 [Ubuntu.1804.Arm64.Open - arm64 - Release (a6fc20e5-6fcc-4e76-a021-83440173fb45)] Will try again after 2 min 0 sec
...
[Pipeline] [Ubuntu.1804.Arm64.Open - arm64 - Release (a6fc20e5-6fcc-4e76-a021-83440173fb45)] echo
14:19:58 [Ubuntu.1804.Arm64.Open - arm64 - Release (a6fc20e5-6fcc-4e76-a021-83440173fb45)] Info: Catastrophic Failure: 6 work items failed
[Pipeline] [Ubuntu.1804.Arm64.Open - arm64 - Release (a6fc20e5-6fcc-4e76-a021-83440173fb45)] }
14:19:59 [Ubuntu.1804.Arm64.Open - arm64 - Release (a6fc20e5-6fcc-4e76-a021-83440173fb45)] Will try again after 2 min 52 sec
...
```

**Comment by maryamariyan** [comment]
Will merge this as the Linux build issue is unrelated.

**Pull request merged**